mmdet/datasets/transforms/transforms.py

```diff
@@ -170,6 +169,189 @@ def __repr__(self) -> str:
+                    interpolation=self.interpolation,
+                    return_scale=True,
+                    backend=self.backend)
+            results['img'] = img
+            results['img_shape'] = img.shape[:2]
+            results['scale_factor'] = (w_scale, h_scale)
+            results['keep_ratio'] = self.keep_ratio
+
+    def _resize_masks(self, results: dict) -> None:
+        """Resize masks with ``results['scale']``"""
+        if results.get('gt_masks', None) is not None:
+            if self.keep_ratio:
+                new_w, new_h = results['keep_ratio_scale']
+                results['gt_masks'] = results['gt_masks'].resize(
+                    (new_h, new_w))
+                results['gt_masks'] = results['gt_masks'].pad(
+                    results['img_shape'], pad_val=self.mask_pad_value)
+            else:
+                results['gt_masks'] = results['gt_masks'].resize(
+                    results['img_shape'])
+
+    def _resize_seg(self, results: dict) -> None:
+        """Resize semantic segmentation map with ``results['scale']``."""
+        if results.get('gt_seg_map', None) is not None:
+            if self.keep_ratio:
+                new_w, new_h = results['keep_ratio_scale']
+                gt_seg = mmcv.imresize(
+                    results['gt_seg_map'], (new_w, new_h),
+                    interpolation='nearest',
+                    backend=self.backend)
+                out_w, out_h = results['scale']
+                padded_gt_seg = np.zeros((out_h, out_w, 3)).astype(np.float32)
+                padded_gt_seg = padded_gt_seg + 255
+                padded_gt_seg[:new_h, :new_w] = gt_seg
+                gt_seg = padded_gt_seg
+            else:
+                gt_seg = mmcv.imresize(
+                    results['gt_seg_map'],
+                    results['scale'],
+                    interpolation='nearest',
+                    backend=self.backend)
+            results['gt_seg_map'] = gt_seg
+
+    @autocast_box_type()
+    def transform(self, results: dict) -> dict:
+        """Transform function to resize images, bounding boxes and semantic
+        segmentation map.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+        Returns:
+            dict: Resized results, 'img', 'gt_bboxes', 'gt_seg_map',
+            'scale', 'scale_factor', 'height', 'width', and 'keep_ratio' keys
+            are updated in result dict.
+        """
+        img = results['img']
+        h, w = img.shape[:2]
+        if self.keep_ratio:
+            scale_factor = min(self.width / w, self.height / h)
+            results['scale_factor'] = (scale_factor, scale_factor)
+
+            real_w, real_h = int(w * float(scale_factor) +
+                                 0.5), int(h * float(scale_factor) + 0.5)
+            results['keep_ratio_scale'] = (real_w, real_h)
+        else:
+            results['scale_factor'] = (self.width / w, self.height / h)
```

Comment:
Also, try to run the image_demo.py with this transform to make sure the inference results are correct.